File: Java/leetcode75/src/test/java/org/phoenix/leetcode/challenges/Problem47_NearestExitFromEntranceInMazeTest.java
package org.phoenix.leetcode.challenges;

import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.assertEquals;

class Problem47_NearestExitFromEntranceInMazeTest {

    private final Problem47_NearestExitFromEntranceInMaze test = new Problem47_NearestExitFromEntranceInMaze();

    @Test
    void nearestExit() {
        char[][] maze = new char[][]{{'+', '+', '.', '+'}, {'.', '.', '.', '+'}, {'+', '+', '+', '.'}};
        int[] entrance = new int[]{1, 2};

        assertEquals(1, test.nearestExit(maze, entrance));
    }
}

File: JavaScript/leetcode75/src/test/Problem47_NearestExitFromEntranceInMaze.test.js
const nearestExit = require("../main/Problem47_NearestExitFromEntranceInMaze");

test("should return nearest exit path", () => {
  const maze = [
      ["+", "+", ".", "+"],
      [".", ".", ".", "+"],
      ["+", "+", "+", "."],
    ],
    entrance = [1, 2];

  expect(nearestExit(maze, entrance)).toBe(1);
});
